kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsIT.java

```diff
@@ -44,35 +60,81 @@ class VaultKmsIT {
     @SuppressWarnings("rawtypes")
-    private VaultContainer vaultContainer;
-    private VaultKms service;
+    private static VaultContainer vaultContainer;
+    private static final String nonTlsService = "nonTls";
+    private static final String pkcs12TrustTlsService = "pkcs12Tls";
+    private static final String filePasswordTlsService = "filePasswordTls";
+    private static final String jksTrustTlsService = "jksTls";
+    private static final String defaultStoreTypeTls = "defaultStoreTypeTls";
+    private static final String insecureService = "tlsInsecure";
+    private static final Map<String, VaultKms> vaultKmsMap = new HashMap<>();
+    private static URI vaultTlsUri;
+
+    private static final CertificateGenerator.Keys keys = CertificateGenerator.generate();
+
+    static List<String> vaultKms() {
+        return vaultKmsMap.keySet().stream().toList();
+    }
 
-    @BeforeEach
-    @SuppressWarnings("resource")
-    void beforeEach() {
+    @BeforeAll
+    public static void beforeAll() {
         assumeThat(DockerClientFactory.instance().isDockerAvailable()).withFailMessage("docker unavailable").isTrue();
+        startVault();
+        String tlsAddress = String.format("https://%s:%s", vaultContainer.getHost(), vaultContainer.getMappedPort(8202));
+        vaultTlsUri = URI.create(tlsAddress);
+        vaultKmsMap.put(nonTlsService, nonTls());
+        vaultKmsMap.put(pkcs12TrustTlsService, getTlsVaultKms(tlsForTrustStoreInlinePassword(keys.pkcs12ClientTruststore())));
+        vaultKmsMap.put(filePasswordTlsService, getTlsVaultKms(tlsForTrustStoreFilePassword(keys.pkcs12ClientTruststore())));
+        vaultKmsMap.put(jksTrustTlsService, getTlsVaultKms(tlsForTrustStoreInlinePassword(keys.jksClientTruststore())));
+        vaultKmsMap.put(defaultStoreTypeTls, getTlsVaultKms(defaultStoreTypeTls(keys.jksClientTruststore())));
+        vaultKmsMap.put(insecureService, getTlsVaultKms(insecureTls()));
+    }
 
+    private static void startVault() {
         vaultContainer = new VaultContainer<>(HASHICORP_VAULT)
                 .withVaultToken(VAULT_TOKEN)
                 .withEnv("VAULT_FORMAT", "json")
+                // in vault server -dev mode the 8200 port cannot be reconfigured to TLS and is also useful so that the
+                // internal vault commands we make while execed into the container can communicate with vault over plain
+                // without having to trust the TLS cert within the container. Plus we can test TLS and non-TLS against the
+                // same container.
+                .withExposedPorts(PLAIN_PORT, TLS_PORT)
+                .withCopyFileToContainer(MountableFile.forClasspathResource("Vault.hcl"), "/vault/config/Vault.hcl")
+                .withCopyFileToContainer(MountableFile.forHostPath(keys.serverCertPem(), 0777), "/vault/config/cert.pem")
+                .withCopyFileToContainer(MountableFile.forHostPath(keys.serverPrivateKeyPem(), 0777), "/vault/config/key.pem")
```

Comment:
I know, I know, its just a test container, but we really shouldn't need write permissions to certificate files!

Reply:
+1

yeah.. before you know it two things happen:  a) code scanning tools go red b) someone copies the code to an app that ends up in production.

Reply:
turns out we didn't need to set it anyway, not sure what changed since I was seeing permissions errors a few days ago. Maybe it's the refactor to bouncycastle and the key files have looser permissions than what was output by openssl and keytool.